src/ExtensionConfigHandler.php

```diff
@@ -352,6 +352,156 @@ protected function getSourceStorageWrapper(array $source_dirs, $subdir = Install
     return $source_storage;
   }
 
+  public function sortAndFilterOutput($dependencies) {
+    // Sort and filter dependency lists, and finally change to simple indexed
+    // array lists. This means the output can then be directly copied for use
+    // in a .info.yml file if the output format is YAML.
+    foreach (array_keys($dependencies) as $dependency_type) {
+      if ($dependencies[$dependency_type]) {
+        sort($dependencies[$dependency_type]);
+        $dependencies[$dependency_type] = array_values($dependencies[$dependency_type]);
+      }
+      else {
+        unset($dependencies[$dependency_type]);
+      }
+    }
+    return $dependencies;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getConfigDependancies($extension, $limit = NULL) {
+    $dependencies = array();
+    $listed_config = $this->getExtensionInfo($extension, 'managed');
+    foreach ($listed_config as $key) {
+      $config_dependencies = $this->getIndividualConfigDependancies($key, $limit);
+      foreach ($config_dependencies as $dependency_type => $type_dependencies) {
+        if (!isset($dependencies[$dependency_type])) {
+          $dependencies[$dependency_type] = array();
+        }
+        $dependencies[$dependency_type] += $type_dependencies;
+      }
+    }
+    // Exclude 'core' and the specified extension from the full list.
+    unset($dependencies['module']['core']);
+    unset($dependencies['module'][$extension]);
+    return $this->sortAndFilterOutput($dependencies);
+  }
+
+  /**
+   * @param $name
+   *   The config key to find the dependencies of.
+   * @param null $recursion_limit
+   *   Optionally limit the levels of recursion.
+   * @return mixed
+   */
+  public function getIndividualConfigDependancies($name, $recursion_limit = NULL) {
```

Comment:
Ok sure - let's add it to the interface then too please.

I guess I had felt like it didn't _really_ do much beyond what core provides easily already, but it's the recursive checking of dependencies that is actually important to us, and is therefore useful enough to be made public. ...and as I'd consider any other implementor to need to implement what we consider useful, I'd put it in the interface - although yeah, not strictly necessary, agreed. 

I recognise I'm being a bit backwards in deciding what should be in the interface by what's in our implementation, but that's all a bit arbitrary anyway as it's a custom service that's unlikely to ever get other implementors. The interface then just becomes a useful place to 'define' our service without being mixed with the actual implementation code.